src/internal.c

```diff
@@ -185,8 +185,13 @@ WOLFSSL_CALLBACKS needs LARGE_STATIC_BUFFERS, please add LARGE_STATIC_BUFFERS
 
 #endif /* !WOLFSSL_NO_TLS12 */
 
-#ifndef NO_WOLFSSL_SERVER
-    #if defined(HAVE_SESSION_TICKET) && !defined(WOLFSSL_NO_DEF_TICKET_ENC_CB)
+#if !defined(NO_WOLFSSL_SERVER) && defined(HAVE_SESSION_TICKET)
+    #if defined(WOLFSSL_HAPROXY)
```

Comment:
Should we support a way to use `initial_ctx` with a macro besides `WOLFSSL_HAPROXY`?

Reply:
Seems shortsighted to limit that area of code to HAPROXY only. Could you instead create a new macro like WOLFSSL_TICKET_USE_INITIAL and set this macro in settings.h for `WOLFSSL_HAPROXY`? At least giving other OSP projects the opportunity to use it?

Reply:
I believe the use case for `initial_ctx` is quite rare, and I'm skeptical about its use in other OSP projects. 
Currently,` #define SSL_TICKET_CTX(ssl) ssl->initial_ctx->ticketEncCtx` and `initial_ctx`, `ssl->initial_ctx` are gated by WOLFSSL_HAPROXY.  Are you suggesting alterations to the gating for all of these?

Reply:
Correct, looking for flexibility around the use of the `initial_ctx` feature. Its a fairly easy refactor. If you think HAPROXY is the only project that will ever use this feature then we can take the PR as-is.

Reply:
Thanks for clarifying. Currently, I'm inclined to leave it as-is and consider the easy refactor only when necessary. None of our supported OSP projects require it presently, and if they had, we would have run into issues earlier. I can't predict the needs of future OSP projects. Although introducing a new setting is straightforward, it implies the added responsibility of maintenance and instructing non-configure autotool users to add one more setting to their user_settings.h file.